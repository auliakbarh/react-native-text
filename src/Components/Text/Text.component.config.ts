const displayName: string = 'Text';

export default {
  displayName,
};
