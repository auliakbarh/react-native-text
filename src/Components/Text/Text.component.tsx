import React from 'react';
import { Text as RNText } from 'react-native';

import config from './Text.component.config';
import type { Props } from './Text.component.types';

const Text: React.FC<Props> = ({ children, ...props }) => {
  return <RNText {...props}>{children}</RNText>;
};

Text.displayName = config.displayName;

export default Text;
