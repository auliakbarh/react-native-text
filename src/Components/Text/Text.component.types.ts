import type { TextProps } from 'react-native';

export interface Props extends TextProps {}
