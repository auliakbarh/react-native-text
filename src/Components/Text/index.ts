export { default } from './Text.component';
export type { Props } from './Text.component.types';
