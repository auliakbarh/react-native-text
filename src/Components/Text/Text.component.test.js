import test from 'react-test-renderer';

import Text from './Text.component';

describe('Text component unit test', () => {
  it('should render Text component with children "Hello World!"', () => {
    const tree = test.create(
      <Text testID="text-hello-world">Hello World!</Text>
    );

    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.toTree().props.children).toBe('Hello World!');
    expect(tree.toTree().props.testID).toBe('text-hello-world');
  });
});
