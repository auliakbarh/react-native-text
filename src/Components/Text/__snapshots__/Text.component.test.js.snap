// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component unit test should render Text component with children "Hello World!" 1`] = `
<Text
  testID="text-hello-world"
>
  Hello World!
</Text>
`;
